src/Hangfire.PostgreSql/ExpirationManager.cs

```diff
@@ -78,36 +78,30 @@ public void Execute(CancellationToken cancellationToken)
             {
                 Logger.DebugFormat("Removing outdated records from table '{0}'...", table);
 
-                int removedCount = 0;
-
-                do
+                UseConnectionDistributedLock(_storage, connection =>
                 {
-                    UseConnectionDistributedLock(_storage, connection =>
+                    int removedCount;
+
+                    do
                     {
-                        using (var transaction = connection.BeginTransaction(IsolationLevel.ReadCommitted))
-                        {
-                            removedCount = connection.Execute(
-                                string.Format(@"
+                        removedCount = connection.Execute(
+                            string.Format(@"
 DELETE FROM """ + _storage.Options.SchemaName + @""".""{0}"" 
 WHERE ""id"" IN (
     SELECT ""id"" 
     FROM """ + _storage.Options.SchemaName + @""".""{0}"" 
     WHERE ""expireat"" < NOW() AT TIME ZONE 'UTC' 
     LIMIT {1}
-)", table, _storage.Options.DeleteExpiredBatchSize.ToString(CultureInfo.InvariantCulture)), transaction);
+)", table, _storage.Options.DeleteExpiredBatchSize.ToString(CultureInfo.InvariantCulture)));
```

Comment:
Why did you put this outside the transaction?

Reply:
A honest whoops of trial and error, I think. Will bring it back tomorrow.

Reply:
Apparently transaction was never actually used. It was passed as query parameter and not as transaction:
![image](https://user-images.githubusercontent.com/12134941/140481692-9ca3fc4b-4b34-4e87-be43-bd51dc09b0b4.png)
Anyway, fixed that as well.